ClickHouse.Client/ADO/ClickHouseCommand.cs

```diff
@@ -52,6 +54,12 @@ public ClickHouseCommand(ClickHouseConnection connection)
 
     public QueryStats QueryStats { get; private set; }
 
+    /// <summary>
+    /// Gets collection of custom settings which will be passed as URL query string parameters.
+    /// </summary>
+    /// <remarks>Not thread-safe.</remarks>
+    public IDictionary<string, object> CustomSettings => customSettings ??= new Dictionary<string, object>();
```

Comment:
An autoproperty would be more efficient and idiomatic here:
```
    public IDictionary<string, object> CustomSettings { get; } = new Dictionary<string, object>();
```

Reply:
You're right, the auto-property is more idiomatic. But is it truly more efficient here? Since commands are created frequently and settings are used only occasionally, I chose not to allocate the dictionary every time to reduce command initialization time.

Reply:
The cost of having an NRE at the wrong time will definitely be more than the cost of creating an extra Dictionary object